pkg/resource/monitoring_schedule/testdata/test_suite.yaml

```diff
@@ -1,17 +1,168 @@
+        error: resource is in terminal condition
+    - name: "Create=Valid"
+      description: "Create a new monitoring schedule successfully (ARN in status)."
+      given:
+        desired_state: "v1alpha1/create/desired/success_before_create.yaml"
+        svc_api:
+          - operation: CreateMonitoringScheduleWithContext
+            output_fixture: "sdkapi/create/success_create.json"
+      invoke: Create
+      expect:
+        latest_state: "v1alpha1/create/observed/success_after_create.yaml"
+        error: null
+  - name: "Monitoring Schedule readOne tests"
+    description: "Testing the readOne operation"
+    scenarios:
+    - name: "ReadOne=MissingRequiredField"
+      description: "Testing readOne when required field is missing. No API call is made and returns error."
+      given: 
+        desired_state: "v1alpha1/readone/desired/missing_required_field.yaml"
+      invoke: ReadOne
+      expect:
+        error: "resource not found"
+    - name: "ReadOne=NotFound"
+      description: "Testing readOne when Describe fails to find the resource on SageMaker"
+      given: 
+        desired_state: "v1alpha1/create/observed/success_after_create.yaml"
+        svc_api:
+          - operation: DescribeMonitoringScheduleWithContext
+            error:
+              code: ResourceNotFound
+              message: "Could not find monitoring schedule"
+      invoke: ReadOne
+      expect:
+        error: "resource not found"
+    - name: "ReadOne=Fail"
+      description: "This test checks if the condition is updated if describe fails and readOne returns error"
+      given: 
+        desired_state: "v1alpha1/create/observed/success_after_create.yaml"
+        svc_api:
+          - operation: DescribeMonitoringScheduleWithContext
+            error:
+              code: ServiceUnavailable
+              message: "Server is down"
+      invoke: ReadOne
+      expect:
+        latest_state: "v1alpha1/readone/observed/error_on_describe.yaml"
+        error: "ServiceUnavailable: Server is down\n\tstatus code: 0, request id: "
+    - name: "ReadOne=AfterCreate"
+      description: "Testing readOne right after create, there should be no delta."
+      given:
+        desired_state: "v1alpha1/create/observed/success_after_create.yaml"
+        svc_api:
+          - operation: DescribeMonitoringScheduleWithContext
+            output_fixture: "sdkapi/describe/success_describe.json"
+      invoke: ReadOne
+      expect:
+        latest_state: "v1alpha1/readone/observed/scheduled.yaml"
+    - name: "ReadOne=SuccessClearsConditions"
+      description: Testing a successful reconciliation clears conditions if terminal/recoverable condition were already set to true
+      given:
+        desired_state: "v1alpha1/readone/desired/error_conditions_true.yaml"
+        svc_api:
+          - operation: DescribeMonitoringScheduleWithContext
+            output_fixture: "sdkapi/describe/success_describe.json"
```

Comment:
why does it reconcile if terminal state is true ? 
(same question for other PRs)